### PR TITLE
Fix unreleased regression on building GroupContact clause

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Adds a unit test & fixes an unreleased regression caused by https://github.com/civicrm/civicrm-core/pull/12344

Before
----------------------------------------
A fix to searching groups by not in caused acl users to not be able to see people in groups they can access.

After
----------------------------------------
Regression fixed & tested

Technical Details
----------------------------------------
There is a lot of complexity in the ACL::whereClause function to add smart group tables in but is also unecessary IMHO since we basically just need to know if the contacts are in the specified groups.  I don't have a good way to performance test this change but it is consistent with changes elsewhere that perform well. Unit test added

Comments
----------------------------------------

### CODE REVIEW

File: tests/phpunit/api/v3/ACLPermissionTest.php [hunk 1]
```diff
@@ -664,4 +664,15 @@ protected function getActivityContacts($activity) {
...
 
+  /**
+   * Test that the 'everyone' group can be given access to a contact.
+   */
+  public function testGetACLEveryonePermittedEntity() {
+    $this->setupScenarioCoreACLEveryonePermittedToGroup();
+    $this->callAPISuccess('Contact', 'getsingle', [
+      'id' => $this->scenarioIDs['Contact']['permitted_contact'],
+      'check_permissions' => 1,
+    ]);
```
Comment (seamuslee001) [inline tests/phpunit/api/v3/ACLPermissionTest.php]: @eileenmcnaughton how about adding this is in
```suggestion
    ]);
    // Check that only permitted contacts can be seen
    $nonPermissionResult = $this->callAPISuccess('Contact', 'get', [
      'id' => $this->scenarioIDs['Contact']['non_permitted_contact'],
      'check_permissions' => 1,
    ]);
    $this->assertTrue(empty($nonPermissionResult['values']));
```

Comment (eileenmcnaughton) [reply tests/phpunit/api/v3/ACLPermissionTest.php]: good idea

**Comment by seamuslee001** [comment]
The changes look fine and i have reviewed the unit test and it looks fine, Just would like thoughts on my possible improvement @eileenmcnaughton otherwise i think this is good

File: tests/phpunit/api/v3/ACLPermissionTest.php [hunk 2]
```diff
@@ -664,4 +664,20 @@ protected function getActivityContacts($activity) {
...
+  /**
+   * Test that the 'everyone' group can be given access to a contact.
+   */
+  public function testGetACLEveryonePermittedEntity() {
+    $this->setupScenarioCoreACLEveryonePermittedToGroup();
+    $this->callAPISuccessGetCount('Contact', [
+      'id' => $this->scenarioIDs['Contact']['permitted_contact'],
+      'check_permissions' => 1,
+    ], 1);
+    $this->setupScenarioCoreACLEveryonePermittedToGroup();
```
Comment (seamuslee001) [inline tests/phpunit/api/v3/ACLPermissionTest.php]: @eileenmcnaughton why need the 2nd call here?

**Comment by eileenmcnaughton** [comment]
I've just re-pushed but can't run locally right now for unknown local reasons

**Comment by seamuslee001** [comment]
I've added merge on pass as i confirmed the Unit test shows the failure

File: tests/phpunit/api/v3/ACLPermissionTest.php (hunk 2 above)
Comment (eileenmcnaughton) [reply tests/phpunit/api/v3/ACLPermissionTest.php]: gone

**Pull request merged**